cmake/Modules/FindSystemBLAS.cmake

```diff
@@ -27,23 +26,42 @@ set(SystemBLAS_FOUND FALSE)
 
 include(FindPackageHandleStandardArgs)
 
-find_library(OPENBLAS_LIBRARIES NAMES openblas libopenblas)
-find_path(OPENBLAS_INCLUDE_DIRS openblas_config.h)
-if(OPENBLAS_LIBRARIES AND OPENBLAS_INCLUDE_DIRS)
+# Let user possibility to set blas path to similar libraries like lapack
+if(BLAS_LIBRARIES AND BLAS_INCLUDE_DIRS)
   find_package(Threads REQUIRED)
   add_library(blas::blas UNKNOWN IMPORTED)
   set_target_properties(blas::blas PROPERTIES
-    INTERFACE_INCLUDE_DIRECTORIES "${OPENBLAS_INCLUDE_DIRS}"
+    INTERFACE_INCLUDE_DIRECTORIES "${BLAS_INCLUDE_DIRS}"
     INTERFACE_LINK_LIBRARIES Threads::Threads
-    IMPORTED_LOCATION "${OPENBLAS_LIBRARIES}"
+    IMPORTED_LOCATION "${BLAS_LIBRARIES}"
   )
-  set(OPENBLAS_FOUND TRUE)
-  set(SystemBLAS_LIBRARIES OPENBLAS_LIBRARIES)
+  set(BLAS_FOUND TRUE)
+  set(SystemBLAS_LIBRARIES USER_DEFINED)
 else()
+  set(BLA_VENDOR OpenBLAS)
   find_package(BLAS QUIET)
+  if(BLAS_FOUND)
+    set(OPENBLAS_FOUND TRUE)
```

Comment:
Oh this is what it was, we should dereference the variable then like ${BLAS_LIBRARIES} so it will print the path of the library that it finds. I think this is more useful than `USER_DEFINED`, etc.